torch/fx/subgraph_rewriter.py

```diff
@@ -220,6 +235,9 @@ def overlaps_with_prev_match(match : Match) -> bool:
                 return True
         return False
 
+    assert matches, ("No matches found between original nn.Module `gm` "
+                     "and pattern `pattern`")
```

Comment:
I don't think this should be an error, it would break composability, e.g. if you want to run a pipeline of replacements that may or may not appear at each stage, this would break that workflow

Reply:
If the user wants to verify that something was matched, we should expose that directly rather than constraining the use cases this can be used for